lib/three_scale/domain_substitution.rb

```diff
@@ -176,9 +169,6 @@ def match_internal_domain?(host)
     end
 
     deprecate domain: "use #internal_domain or #external_domain",
-      self_domain: "use #internal_admin_domain or #external_admin_domain",
-      external_self_domain: "use #external_admin_domain",
```

Comment:
I meant `#external_self_domain` because it still exists. `self_domain` is perhaps the attribute. We can't remove it, probably we can make it private at most before removing it from deprecation.